cmd/multiFaAcceleration/multiFaAcceleration.go

```diff
@@ -0,0 +1,504 @@
+func pruneRight(d Distances, b BranchLengths, sub *SubTree, s Settings) {
+	sub.Dac = d.D23
+
+	if d.D02 == 0 {
+		if d.D12 == 0 {
+			sub.Dac = (s.ZeroDistanceWeightConstant*(d.D02-b.B1) + s.ZeroDistanceWeightConstant*(d.D12-b.B2)) / (2.0 * s.ZeroDistanceWeightConstant)
+		} else {
+			sub.Dac = ((1.0/math.Pow(d.D12, 2))*(d.D12-b.B2) + s.ZeroDistanceWeightConstant*(d.D02-b.B1)) / ((1.0 / math.Pow(d.D12, 2)) + s.ZeroDistanceWeightConstant)
+		}
+	} else if d.D12 == 0 {
+		sub.Dac = ((1.0/math.Pow(d.D02, 2))*(d.D02-b.B1) + s.ZeroDistanceWeightConstant*(d.D12-b.B2)) / ((1.0 / math.Pow(d.D02, 2)) + s.ZeroDistanceWeightConstant)
+	} else {
+		sub.Dab = ((1.0/math.Pow(d.D02, 2))*(d.D02-b.B1) + (1.0/math.Pow(d.D12, 2))*(d.D12-b.B2)) / ((1.0 / math.Pow(d.D02, 2)) + (1.0 / math.Pow(d.D12, 2)))
+	}
+
+	if d.D03 == 0 {
+		if d.D13 == 0 {
+			sub.Dbc = (s.ZeroDistanceWeightConstant*(d.D03-b.B1) + s.ZeroDistanceWeightConstant*(d.D13-b.B2)) / (2.0 * s.ZeroDistanceWeightConstant)
+		} else {
+			sub.Dbc = ((1.0/math.Pow(d.D13, 2))*(d.D13-b.B2) + s.ZeroDistanceWeightConstant*(d.D03-b.B1)) / ((1.0 / math.Pow(d.D13, 2)) + s.ZeroDistanceWeightConstant)
+		}
+	} else if d.D13 == 0 {
+		sub.Dbc = ((1.0/math.Pow(d.D03, 2))*(d.D03-b.B1) + s.ZeroDistanceWeightConstant*(d.D13-b.B2)) / ((1.0 / math.Pow(d.D03, 2)) + s.ZeroDistanceWeightConstant)
+	} else {
+		sub.Dbc = ((1.0/math.Pow(d.D03, 2))*(d.D03-b.B1) + (1.0/math.Pow(d.D13, 2))*(d.D13-b.B2)) / ((1.0 / math.Pow(d.D03, 2)) + (1.0 / math.Pow(d.D13, 2)))
+	}
+}
+
+//For a set of distances and corresponding branch lengths, determine the value of Q, the Fitch-Margoliash least squares error.
+func calculateQ(d Distances, b BranchLengths, s Settings) float64 {
+	var sum float64 = 0
+	if d.D01 != 0 { //avoid divide by zero error
+		sum += math.Pow(d.D01-b.B1-b.B2, 2) / math.Pow(d.D01, 2)
+	} else {
+		sum += math.Pow(d.D01-b.B1-b.B2, 2) * s.ZeroDistanceWeightConstant
+	}
+	if d.D02 != 0 {
+		sum += math.Pow(d.D02-b.B1-b.B3-b.B4, 2) / math.Pow(d.D02, 2)
+	} else {
+		sum += math.Pow(d.D02-b.B1-b.B3-b.B4, 2) * s.ZeroDistanceWeightConstant
+	}
+	if d.D03 != 0 {
+		sum += math.Pow(d.D03-b.B1-b.B3-b.B5, 2) / math.Pow(d.D03, 2)
+	} else {
+		sum += math.Pow(d.D03-b.B1-b.B3-b.B5, 2) * s.ZeroDistanceWeightConstant
+	}
+	if d.D12 != 0 {
+		sum += math.Pow(d.D12-b.B2-b.B3-b.B4, 2) / math.Pow(d.D12, 2)
+	} else {
+		sum += math.Pow(d.D12-b.B2-b.B3-b.B4, 2) * s.ZeroDistanceWeightConstant
+	}
+	if d.D13 != 0 {
+		sum += math.Pow(d.D13-b.B2-b.B3-b.B5, 2) / math.Pow(d.D13, 2)
+	} else {
+		sum += math.Pow(d.D13-b.B2-b.B3-b.B5, 2) * s.ZeroDistanceWeightConstant
+	}
+	if d.D23 != 0 {
+		sum += math.Pow(d.D23-b.B4-b.B5, 2) / math.Pow(d.D23, 2)
+	} else {
+		sum += math.Pow(d.D23-b.B4-b.B5, 2) * s.ZeroDistanceWeightConstant
+	}
+	return sum
+}
+
+//bitArray is on reference coordinates, not alignment coordinates, so the window is simply equal to windowSize.
```

Comment:
Could you help me understand the difference between reference coordinates and alignment coordinates?